ta4j-core/src/main/java/org/ta4j/core/indicators/helpers/CriterionIndicator.java

```diff
@@ -0,0 +1,124 @@
+        this.tradingRecord = tradingRecord;
+        this.criterion = criterion;
+        this.requiredCriterionValue = requiredCriterionValue;
+        this.position = null;
+    }
+
+    /**
+     * Constructor.
+     * 
+     * @param series                 the bar series
+     * @param position               the position
+     * @param criterion              the criterion to get the calculated criterion
+     *                               value on a bar index
+     * @param requiredCriterionValue the required criterion value to test against
+     *                               the calculated criterion value
+     */
+    public CriterionIndicator(BarSeries series, Position position, AnalysisCriterion criterion,
+            Num requiredCriterionValue) {
+        super(series);
+        this.tradingRecord = null;
+        this.criterion = criterion;
+        this.requiredCriterionValue = requiredCriterionValue;
+        this.position = position;
+    }
+
+    @Override
+    protected Boolean calculate(int index) {
+        if (tradingRecord != null && !tradingRecord.getPositions().isEmpty()) {
+            TradingRecord tradingRecordTillIndex = new BaseTradingRecord(tradingRecord.getStartingType(),
+                    tradingRecord.getTransactionCostModel(), tradingRecord.getHoldingCostModel());
+            for (Position pos : tradingRecord.getPositions()) {
+                // consider only those positions made till index
+                if (pos.getEntry().getIndex() <= index) {
+                    Trade entry = pos.getEntry();
+                    tradingRecordTillIndex.enter(entry.getIndex(), entry.getNetPrice(), entry.getAmount());
+                }
+                if (pos.getExit() != null && pos.getExit().getIndex() <= index) {
+                    Trade exit = pos.getExit();
+                    tradingRecordTillIndex.exit(exit.getIndex(), exit.getNetPrice(), exit.getAmount());
+                }
+            }
+
+            Num calculatedCriterionValue = criterion.calculate(getBarSeries(), tradingRecordTillIndex);
+            return criterion.betterThan(calculatedCriterionValue, requiredCriterionValue)
+                    || calculatedCriterionValue.isEqual(requiredCriterionValue);
+
+        } else if (position != null) {
+            boolean entryTradeWithinIndex = position.getEntry().getIndex() <= index;
+            boolean exitTradeWithinIndex = position.getExit() == null
+                    || position.getExit() != null && position.getExit().getIndex() <= index;
+            if (entryTradeWithinIndex && exitTradeWithinIndex) {
+                Num calculatedCriterionValue = criterion.calculate(getBarSeries(), position);
+                return criterion.betterThan(calculatedCriterionValue, requiredCriterionValue)
+                        || calculatedCriterionValue.isEqual(requiredCriterionValue);
+            }
+        }
+        // empty tradingRecords or no position must return undefined
+        return false;
```

Comment:
The questions is: 
Should I return `null` instead of `false` ? The null (for Boolean) is similar to NaN for numbers. As long as **no** trades are recorded, neither true nor false can be determined, thus null. 

@team172011 what do you think?

Reply:
I would return false, I do not like the possibility of trivalent expressions for boolean values